plugins/vercel/api_token.go

```diff
@@ -0,0 +1,87 @@
+package vercel
+
+import (
+	"context"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func APIToken() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.APIToken,
+		DocsURL:       sdk.URL("https://vercel.com/docs/cli/login"),
+		ManagementURL: sdk.URL("https://vercel.com/account/tokens"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.Token,
+				MarkdownDescription: "Token used to authenticate to Vercel.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 24,
+					Charset: schema.Charset{
+						Lowercase: true,
+						Uppercase: true,
+						Digits:    true,
+					},
+				},
+			},
+		},
+		DefaultProvisioner: vercelProvisioner{},
+		Importer: importer.TryAll(
+			importer.TryEnvVarPair(
+				map[string]sdk.FieldName{
+					"VERCEL_TOKEN": fieldname.Token,
+				},
+			),
+			importer.MacOnly(TryVercelConfigFile("~/Library/Application Support/com.vercel.cli/auth.json")),
+			importer.LinuxOnly(TryVercelConfigFile("~/.config/com.vercel.cli/auth.json")),
+		),
+	}
+}
+
+type vercelProvisioner struct{}
+
+func (v vercelProvisioner) Description() string {
+	return "Vercel cli token provisioner"
+}
+
+func (v vercelProvisioner) Provision(ctx context.Context, input sdk.ProvisionInput, output *sdk.ProvisionOutput) {
```

Comment:
Is there a reason you've opted for a file-based provisioner, instead of an env-var provisioner? 

Generally, I suggest using an env-var provisioner where possible, due to current limitations of the file-based approach when it comes to multiple reads within the same command.

Reply:
Actually I used a arguments-based provisioner (providing `--token` argument). I checked the documentation and [source code](https://github.com/vercel/vercel/tree/main/packages/cli) for the vercel-cli, but I couldn't find any evidence that it reads tokens from environment variables.

While we could use a file-based provisioner instead, I think that using the `--token` argument is the simplest approach.

Reply:
My apologies, slightly misread the code! I think the initial version of the code was also importing from `VERCEL_TOKEN`, reason why I assumed it would work. This is fine, in this situation. Thanks for clarifying!